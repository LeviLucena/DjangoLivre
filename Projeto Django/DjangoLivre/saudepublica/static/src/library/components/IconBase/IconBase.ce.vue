<template>
  <i :class="iconClasses" aria-hidden="true"></i>
</template>

<script>
const IconBase = {
  name: "IconBase",
  props: {
    /**
     * Nome do ícone que será apresentado
     */
    iconName: {
      type: String,
      default: "",
      validator: function (val) {
        return val.trim().length > 1
      },
    },
    /**
     * Se informado define que o nome da familia de fonte que será aplicada ao ícone
     */
    familyName: {
      type: String,
      default: "fas",
      validator: function (val) {
        return val.trim().length > 1
      },
    },
    /**
     * Se informado define qual o estilo de classe css que será aplicada ao ícone
     */
    className: {
      type: [String, Object],
      default: null,
    },
  },
  computed: {
    /**
     * Retorna o nome do ícone que será aplicado ao componente
     */
    iconClasses() {
      return [this.familyName, `fa-${this.processIconName(this.iconName)}`, this.className].join(" ")
    },
  },
  methods: {
    /**
     * Retorna o nome processado do ícone que será
     * apresentado
     */
    processIconName(iconName) {
      // Remove o prefixo "fa-" se estiver presente
      return iconName.replace(/^fa-/, "")
    },
  },
}

export default IconBase
</script>

<style lang="scss">
@import "~@govbr-ds/core/src/partial/scss/_base";
@import "~@govbr-ds/core/src/partial/scss/_utilities";
@import url("https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.2/css/all.min.css");
</style>
