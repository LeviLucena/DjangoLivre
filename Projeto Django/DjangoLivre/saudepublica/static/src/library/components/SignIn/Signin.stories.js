import { kebabiseArgs } from '../../util/Utils.js'
import Signin from './SignIn.ce.vue'

export default {
  title: 'Componentes/Signin',
  component: Signin,
  parameters: {
    actions: {
      handles: ['click'],
    },
  },
  argTypes: {
    label: {
      control: 'text',
      defaultValue: null,
    },
    type: {
      control: {
        type: 'inline-radio',
        options: [null, 'primary'],
        labels: {
          null: 'Secundário',
          primary: 'Primário',
        },
      },
      defaultValue: '',
    },
    density: {
      control: 'inline-radio',
      options: ['large', 'middle', 'small'],
      defaultValue: 'middle',
    },
    block: {
      defaultValue: false,
    },
    icon: {
      defaultValue: null,
    },
    inverted: {
      defaultValue: false,
    },
    image: {
      defaultValue: null,
    },
    entity: {
      defaultValue: null,
    },
    tooltipText: {
      control: 'text',
    },
    tooltipPlace: {
      options: ['top', 'right', 'bottom', 'left'],
      control: 'inline-radio',
      if: { arg: 'tooltipText', truthy: true },
    },
    tooltipType: {
      options: ['info', 'success', 'warning', 'error'],
      control: 'inline-radio',
      if: { arg: 'tooltipText', truthy: true },
    },
    tooltipTimer: {
      control: 'number',
      if: { arg: 'tooltipText', truthy: true },
      isLink: {
        defaultValue: null,
      },
    },
    href: {
      control: 'text',
      if: { arg: 'isLink' },
      defaultValue: '#url',
    },
  },
}

const Template = (args) => ({
  props: Object.keys(args),
  setup() {
    return { args: kebabiseArgs(args) }
  },
  template: `<br-sign-in v-bind="args"></br-sign-in>`,
})

const TemplateFundoEscuro = (args) => ({
  props: Object.keys(args),
  setup() {
    return { args: kebabiseArgs(args) }
  },
  template: `
  <div class="p-3 bg-primary-darken-02">
    <br-sign-in v-bind="args"></br-sign-in>
  </div>`,
})

export const TipoInterno = Template.bind({})
TipoInterno.args = {
  type: 'primary',
  density: 'middle',
  icon: 'user',
  label: 'Entrar',
}

export const TipoExternoComTexto = Template.bind({})
TipoExternoComTexto.args = {
  type: 'primary',
  density: 'middle',
  label: 'Entrar com',
  entity: 'gov.br',
}

export const TipoExternoComImagem = Template.bind({})
TipoExternoComImagem.args = {
  type: '',
  density: 'middle',
  label: 'Entrar com',
  image: "{ 'url': 'https://www.gov.br/++theme++padrao_govbr/img/govbr-colorido-b.png', 'description': 'gov.br' }",
}

export const Iconico = Template.bind({})
Iconico.args = {
  type: 'primary',
  density: 'large',
  icon: 'user',
}

export const EnfasePrimariaComFundoEscuro = TemplateFundoEscuro.bind({})
EnfasePrimariaComFundoEscuro.args = {
  type: 'primary',
  inverted: true,
  density: 'middle',
  label: 'Entrar com',
  image: "{ 'url': 'https://www.gov.br/++theme++padrao_govbr/img/govbr-colorido-b.png', 'description': 'gov.br' }",
}
EnfasePrimariaComFundoEscuro.parameters = {
  controls: {
    exclude: ['iconic'],
  },
}

export const Bloco = Template.bind({})
Bloco.args = {
  type: 'primary',
  density: 'middle',
  icon: 'user',
  label: 'Entrar',
  block: true,
}

export const ComTooltip = Template.bind({})
ComTooltip.args = {
  type: 'primary',
  density: 'large',
  icon: 'user',
  tooltipText: 'Lorem ipsum dolor sit amet.',
}

export const Hiperlink = Template.bind({})
Hiperlink.args = {
  type: 'primary',
  density: 'middle',
  icon: 'user',
  label: 'Entrar',
  isLink: true,
}
