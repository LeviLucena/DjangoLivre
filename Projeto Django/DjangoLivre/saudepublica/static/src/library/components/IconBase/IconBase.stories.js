import { kebabiseArgs } from '../../util/Utils.js'
import IconBase from './IconBase.ce.vue'

export default {
  title: 'Componentes/Icon',
  component: IconBase,
  argTypes: {
    iconName: {
      control: {
        type: 'text',
      },
    },
    familyName: {
      control: {
        type: 'text',
      },
    },
    className: {
      control: {
        type: 'text',
      },
    },
  },
}

const Template = (args) => ({
  props: Object.keys(args),
  setup() {
    return { args: kebabiseArgs(args) }
  },
  template: `<br-icon-base v-bind="args"></br-icon-base>`,
})

export const Padrão = Template.bind({})
Padrão.args = {
  iconName: 'home',
  familyName: 'fas',
  className: '',
}
