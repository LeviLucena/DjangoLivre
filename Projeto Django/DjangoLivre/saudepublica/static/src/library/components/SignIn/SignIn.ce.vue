<template>
  <BrButton
    v-if="!isLink"
    v-tooltip="{ text: tooltipText, place: tooltipPlace, type: tooltipType, timer: tooltipTimer }"
    class="sign-in"
    :type="type"
    :density="density"
    :inverted="inverted"
    :circle="_iconic"
    :block="block"
  >
    <IconBase
      v-if="_iconic || (icon && label && !image && !entity)"
      :icon-name="icon"
      :class-name="icon && label ? 'mr-1' : ''"
    ></IconBase>
    <span v-if="label">{{ label }}</span>
    <span v-if="entity && !image" class="ml-1 text-black">{{ entity }}</span>
    <img v-else-if="image" class="ml-1" :src="_image.url" :alt="_image.description" />
  </BrButton>
  <a v-else v-bind="$attrs" class="br-sign-in" :class="classObject" :href="href">
    <IconBase
      v-if="_iconic || (icon && label && !image && !entity)"
      :icon-name="icon"
      :class-name="icon && label ? 'mr-1' : ''"
    ></IconBase>
    <span v-if="label">{{ label }}</span>
    <span v-if="entity && !image" class="ml-1 text-black">{{ entity }}</span>
    <img v-else-if="image" class="ml-1" :src="_image.url" :alt="_image.description" />
  </a>
</template>

<script>
import TooltipMixin from "../../mixins/TooltipMixin"
import { parsePropStringToJSON } from "../../util/Utils.js"
import BrButton from "../Button/Button.ce.vue"
import IconBase from "../IconBase/IconBase.ce.vue"

const SignIn = {
  name: "brSignIn",
  components: {
    BrButton,
    IconBase,
  },
  mixins: [TooltipMixin],
  props: {
    /**
     * **[OPCIONAL]** Ênfase do sign-in, com finalidade de proporcionar mais ou menos destaque em relação a outros componentes exibidos na tela. A ênfase secundaria é padrão quando o type não é passado.
     */
    type: {
      type: String,
      validator: function (value) {
        return ["primary"].includes(value)
      },
    },
    /** **[OPCIONAL]** O ajuste da densidade consiste em aumentar ou reduzir os espaçamentos internos do sign-in de forma a deixá-los mais compactos ou mais expandidos em seu formato. **[ large | middle | small ]**
     */
    density: {
      type: String,
      validator: function (value) {
        return ["large", "middle", "small"].includes(value)
      },
      default: "middle",
    },
    /**
     * **[OPCIONAL]** Usado quando houver uma necessidade de enfatizar visualmente a ideia da ação. O ícone de usuário deverá ser localizado à esquerda do rótulo, diferentemente do sign-in to tipo externo que não possui ícone e a sua imagem localiza-se à direita do rótulo.
     */
    icon: {
      type: String,
      default: null,
    },
    /**
     * **[OBRIGATÓRIO]** O rótulo é um elemento obrigatório. A **exceção** ocorre na versão tipo icônico uma vez que não existe rótulo.
     */
    label: {
      type: String,
      default: null,
    },
    /**
     * **[OPCIONAL]** Objeto que representa a imagem destinada à utilização de uma
     * assinatura ou logo institucional do SignIn.
     *
     * O objeto deve seguir a estrutura abaixo:
     *
     * ```
     * {
     *   // [OBRIGATÓRIO] URL da imagem
     *   name: 'string',
     *
     *   // [OBRIGATÓRIO] Descrição alternativa da imagem
     *   description: 'string'
     * }
     * ```
     *
     * Quando passado como atributo, obrigatoriamente deve ser um objeto convertido
     * para string. Quando passado como propriedade, pode ser um objeto propriamente
     * dito.
     *
     * Referência: https://open-wc.org/guides/knowledge/attributes-and-properties
     */
    image: {
      type: [String, Object],
      default: null,
    },
    /**
     * **[OPCIONAL]** Nome institucional utilizado em substituição à imagem.
     */
    entity: {
      type: String,
      default: null,
    },
    /**
     * **[OPCIONAL]** Utilizado quando o background possui cores escuras.
     */
    inverted: {
      type: Boolean,
      default: false,
    },
    /** **[OPCIONAL]** Apresenta o sign-in em comportamento de bloco. o sign-in “empurra” qualquer outro elemento existente para outro espaço disponível. */
    block: {
      type: Boolean,
      default: false,
    },
    /** **[OPCIONAL]** Sign-in definido como hiperlink (button é o padrão), com possibilidade de utilizar todos os seus atributos.*/
    isLink: {
      type: Boolean,
      default: false,
    },
    /** **[OPCIONAL]** Atributo disponível para sign-in do tipo link (is-link).*/
    href: {
      type: String,
      default: null,
    },
  },
  computed: {
    _image() {
      return parsePropStringToJSON(this.image)
    },
    _iconic() {
      return this.icon?.length > 0 && !this.label && !this.image && !this.entity
    },
    classObject() {
      return [
        {
          primary: this.type === "primary",
          secondary: this.type === "secondary",
          large: this.density === "large",
          medium: this.density === "medium",
          small: this.density === "small",
          inverted: this.inverted,
          circle: this._iconic,
          block: this.block,
        },
      ]
    },
  },
}

export default SignIn
</script>

<style lang="scss">
@import "~@govbr-ds/core/src/partial/scss/_base";
@import "~@govbr-ds/core/src/partial/scss/_utilities";
@import "~@govbr-ds/core/src/components/button/_button";
@import "~@govbr-ds/core/src/components/signin/_signin";
@import url("https://cdnjs.cloudflare.com/ajax/libs/font-awesome/5.11.2/css/all.min.css");
</style>
