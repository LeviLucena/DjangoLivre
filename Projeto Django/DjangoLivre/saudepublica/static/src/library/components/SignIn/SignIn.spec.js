import { shallowMount } from '@vue/test-utils'
import BrButton from '../Button/Button.ce.vue'
import IconBase from '../IconBase/IconBase.ce.vue'
import SignIn from './SignIn.ce.vue'

const densidades = ['large', 'middle', 'small']
const tipo = ['primary']

describe('SignIn', () => {
  const wrapperInternalSignIn = shallowMount(SignIn, {
    propsData: {
      icon: 'user',
      label: 'Entrar',
    },
    global: {
      stubs: {
        'icon-base': IconBase,
        'br-button': BrButton,
      },
    },
  })
  test('it renders br-button', () => {
    expect(wrapperInternalSignIn.classes('br-button')).toBe(true)
  })
  test('it renders icon', () => {
    expect(wrapperInternalSignIn.find('.fas.fa-user').exists()).toBe(true)
  })
  test('it renders label', () => {
    expect(wrapperInternalSignIn.text()).toMatch('Entrar')
  })

  const wrapperExternalSignInWithImage = shallowMount(SignIn, {
    propsData: {
      icon: 'user',
      label: 'Entrar com',
      image: "{ 'url': 'https://www.gov.br/++theme++padrao_govbr/img/govbr-colorido-b.png', 'description': 'gov.br' }",
    },
    global: {
      stubs: {
        'icon-base': IconBase,
        'br-button': BrButton,
      },
    },
  })
  test('it renders image', () => {
    expect(
      wrapperExternalSignInWithImage
        .find('img[src="https://www.gov.br/++theme++padrao_govbr/img/govbr-colorido-b.png"]')
        .exists()
    ).toBe(true)
  })
  test('it not renders icon', () => {
    expect(wrapperExternalSignInWithImage.find('.fa-user').exists()).toBe(false)
  })

  const wrapperExternalSignInWithName = shallowMount(SignIn, {
    propsData: {
      icon: 'user',
      label: 'Entrar com',
      image: "{ 'url': 'https://www.gov.br/++theme++padrao_govbr/img/govbr-colorido-b.png', 'description': 'gov.br' }",
      entity: 'gov.br',
    },
    global: {
      stubs: {
        'icon-base': IconBase,
        'br-button': BrButton,
      },
    },
  })
  test('it renders image', () => {
    expect(
      wrapperExternalSignInWithName
        .find('img[src="https://www.gov.br/++theme++padrao_govbr/img/govbr-colorido-b.png"]')
        .exists()
    ).toBe(true)
  })
  test('it not renders entity name', () => {
    expect(wrapperExternalSignInWithName.text()).not.toMatch('gov.br')
  })
  test('it not renders icon', () => {
    expect(wrapperExternalSignInWithName.find('.fa-user').exists()).toBe(false)
  })

  const wrapperIconicSignIn = shallowMount(SignIn, {
    propsData: {
      icon: 'user',
    },
    global: {
      stubs: {
        'icon-base': IconBase,
        'br-button': BrButton,
      },
    },
  })
  test('it has class circle', () => {
    expect(wrapperIconicSignIn.classes('circle')).toBe(true)
  })
  test('it renders icon', () => {
    expect(wrapperIconicSignIn.find('.fa-user').exists()).toBe(true)
  })

  densidades.forEach((densidade) => {
    test(`set density attribute ${densidade}`, () => {
      const wrapperSignInDensity = shallowMount(SignIn, {
        propsData: {
          icon: 'user',
          label: 'Entrar',
          density: densidade,
        },
        global: {
          stubs: {
            'icon-base': IconBase,
            'br-button': BrButton,
          },
        },
      })
      expect(wrapperSignInDensity.find(`.${densidade}`).exists()).toBe(true)
    })
  })

  tipo.forEach((cor) => {
    test(`set type attribute ${cor}`, () => {
      const wrapperSignInType = shallowMount(SignIn, {
        propsData: {
          type: cor,
        },
        global: {
          stubs: {
            'icon-base': IconBase,
            'br-button': BrButton,
          },
        },
      })
      expect(wrapperSignInType.find(`.${cor}`).exists()).toBe(true)
    })
  })

  const wrapperSignInInverted = shallowMount(SignIn, {
    propsData: {
      inverted: true,
      icon: 'user',
      label: 'Entrar',
    },
    global: {
      stubs: {
        'icon-base': IconBase,
        'br-button': BrButton,
      },
    },
  })
  test('it renders inverted', () => {
    expect(wrapperSignInInverted.classes('inverted')).toBe(true)
  })

  const wrapperSignInBlock = shallowMount(SignIn, {
    propsData: {
      block: true,
      icon: 'user',
      label: 'Entrar',
    },
    global: {
      stubs: {
        'icon-base': IconBase,
        'br-button': BrButton,
      },
    },
  })
  test('it renders sign-in block', () => {
    expect(wrapperSignInBlock.find('.block').exists()).toBe(true)
  })

  test('it renders hiperlink instead of button', () => {
    const wrapper = shallowMount(SignIn, {
      propsData: {
        label: 'Entrar',
        isLink: true,
      },
    })
    expect(wrapper.find('a.br-sign-in').exists()).toBe(true)
    expect(wrapper.find('button.br-sign-in').exists()).toBe(false)
    expect(wrapper.element).toMatchSnapshot()
  })
})
