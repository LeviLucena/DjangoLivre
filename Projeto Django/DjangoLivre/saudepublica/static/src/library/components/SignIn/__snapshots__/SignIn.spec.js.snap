// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SignIn it renders hiperlink instead of button 1`] = `
<a
  class="br-sign-in"
>
  <!--v-if-->
  <span>
    Entrar
  </span>
  <!--v-if-->
</a>
`;
